appservice/src/intTest/groovy/nu/yona/server/CreateUserOnBuddyRequestTest.groovy

```diff
@@ -112,6 +114,122 @@ class CreateUserOnBuddyRequestTest extends AbstractAppServiceIntegrationTest
 		appService.deleteUser(updatedBob)
 	}
 
+	def 'Bob adjusts data, including his Android device'()
+	{
+		given:
+		def richard = addRichard()
+		def mobileNumberBob = makeMobileNumber(timestamp)
+		def inviteUrl = buildInviteUrl(sendBuddyRequestForBob(richard, mobileNumberBob))
+		def bob = appService.getUser(appService.&assertUserGetResponseDetailsWithPrivateDataCreatedOnBuddyRequest, inviteUrl, true, null)
+
+		when:
+		def newNickname = "Bobby"
+		def updatedBobJson = bob.convertToJson()
+		updatedBobJson.nickname = newNickname
+		User bobToBeUpdated = new User(updatedBobJson)
+		bobToBeUpdated.deviceName = "My S8"
+		bobToBeUpdated.deviceOperatingSystem = "ANDROID"
+		User updatedBob = appService.updateUserCreatedOnBuddyRequest(appService.&assertUserUpdateResponseDetails, bobToBeUpdated, inviteUrl)
+
+		then:
+		updatedBob.devices.size() == 0 // Mobile number not confirmed yet
+		
+		def bobWithConfirmedNumber = appService.confirmMobileNumber({ AppService.assertResponseStatusSuccess(it)}, updatedBob)
+		bobWithConfirmedNumber.devices.size() == 1
+		bobWithConfirmedNumber.devices[0].name == "My S8"
+		bobWithConfirmedNumber.devices[0].operatingSystem == "ANDROID"
+		
+		cleanup:
+		appService.deleteUser(richard)
+		appService.deleteUser(updatedBob)
+	}
+
+	def 'Bob adjusts data, including his iOS device'()
```

Comment:
I don't think this test makes sense, because it is exactly the same as the previous one, except for some data.

Reply:
I wanted to verify that both supported operating systems indeed work.

Reply:
With the argument that we already test the types of devices in ``UserTest`` (see your next comment), I will leave this test out too.